openedx/core/djangoapps/call_stack_manager/tests.py

```diff
@@ -116,12 +149,18 @@ class TestingCallStackManager(TestCase):
     1. Tests CallStackManager QuerySetAPI functionality
     2. Tests @donottrack decorator
     """
+    def setUp(self):
+        core.TRACK_FLAG = True
+        core.STACK_BOOK = collections.defaultdict(list)
+        core.HALT_TRACKING = []
+
     def test_save(self, log_capt):
         """ tests save() of CallStackMixin/ applies same for delete()
         classes with CallStackMixin should participate in logging.
         """
         ModelMixin(id_field=1).save()
-        self.assertEqual(ModelMixin, log_capt.call_args[0][1])
+        self.assertEqual(ModelMixin, log_capt.call_args[0][2])
+        self.addCleanup(log_capt)
```

Comment:
`addCleanup` takes a function to be called after the test. You should call it in `setUp`, not in the test, and the function that you pass it should perform whatever cleanup needs to happen (like lines 153-155 in the current code).